src/oc_erchef/apps/chef_index/src/chef_index.erl

```diff
@@ -111,15 +104,67 @@ add(TypeName, Id, DbName, IndexEjson, ReqId) ->
             send_to_solr(QueueMode, Doc, ReqId)
     end.
 
-add_async(TypeName, Id, DbName, IndexEjson, _ReqId) ->
-    %% No solr_time logging for add_async as this doesn't happen as
-    %% part of a request, But we provided a add_async/5 to make it
-    %% easier to call in oc_chef_object_db.
-    add_async(TypeName, Id, DbName, IndexEjson).
+-spec add_batch(list()) -> ok | {error, list()}.
+add_batch(Batch) ->
+    {ok, BatchWorker} = chef_wait_group:start_link(fun add_batch_item_with_retries/1, []),
+    [chef_wait_group:add(BatchWorker, {TypeName, Id, DbName}, Item) || {TypeName, Id, DbName, _} = Item <- Batch],
+    case chef_wait_group:wait(BatchWorker) of
+        {ok, Results}  ->
+            case all_ok(Results) of
```

Comment:
We could also pattern match `not_ok(Results)` here and save a call. (Don't think it matters much, though.)